mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulHome.kt

```diff
@@ -81,7 +84,18 @@ fun StatefulHome(
             contentPadding = paddingValues,
         )
       }
-      composable(SettingsRoute) { StatefulSettingsScreen(user, Modifier.fillMaxSize()) }
+      composable(SettingsRoute) {
+        StatefulSettingsScreen(
+            user = user,
+            modifier = Modifier.fillMaxSize(),
+            openEditProfileName = { controller.navigate(SettingEditProfileNameRoute) })
+      }
+      dialog(SettingEditProfileNameRoute) {
+        StatefulEditProfileDialog(
+            user = user,
+            onSave = { controller.navigate(SettingsRoute) },
+            onCancle = { controller.navigate(SettingsRoute) })
```

Comment:
Maybe use `controller.popBackStack()` instead, which may handle pressing the back button better since it removes the dialog destination from the back stack.

Reply:
👌 is changed